src/components/Toast/toastStore.ts

```diff
@@ -0,0 +1,94 @@
+import { nanoid } from 'nanoid';
+import { action, atom, WritableAtom } from 'nanostores';
+import { ToastType } from './enums';
+import { Toast, ToastOptions, ToastUpdateOptions } from './types';
+
+export const toastStore = atom<Toast[]>([]);
+
+export const updateToast = action(
+  toastStore,
+  'updateToast',
+  (
+    store: WritableAtom<Toast[]>,
+    toastId: Toast['id'],
+    options: ToastUpdateOptions
+  ) => {
+    const existingToasts = store.get();
+    store.set(
+      existingToasts.map((toast) => {
+        if (toast.id === toastId) {
+          return {
+            ...toast,
+            ...options,
+          };
+        } else {
+          return toast;
+        }
+      })
+    );
+  }
+);
+
+const removeToast = action(
+  toastStore,
+  'removeToast',
+  (store: WritableAtom<Toast[]>, toastId: Toast['id']) => {
+    const existingToasts = store.get();
+    store.set(existingToasts.filter((toast) => toast.id !== toastId));
+  }
+);
+
+const DELAY_BEFORE_REMOVING_TOAST = 175;
```

Comment:
Perhaps this should be a property of the toast itself and can be configurable by the clients, but defaults to 175?